scalyr_agent/copying_manager/worker.py

```diff
@@ -1036,21 +1043,58 @@ def __scan_for_new_bytes(self, current_time=None):
         for processor in self.__log_processors:
             processor.scan_for_new_bytes(current_time)
 
-    def get_full_checkpoints(self):  # type: () -> Dict[six.text_type, Any]
+    def _update_checkpoints(self, current_time):
+        """
+        Update the checkpoints collection from the current state of the log processors.
+        :param current_time:
+        :return:
+        """
+        # get all log processors, including those that need to be removed.
+        log_processors = self.__log_processors
+
+        with self._checkpoints_lock:
+            # remove stale checkpoints using the same config option as for the whole file staleness check.
+            for path, checkpoint in list(self._checkpoints.items()):
```

Comment:
So, we are changing the workers to also track the checkpoints for files that they have recently processed.  Is that correct?  And we just remove them when they exceed a maximum checkpoint age.

But, we really only need them to hold onto these old checkpoints until the main agent process retrieves this checkpoint and writes it down to disk.  Is that right?